lib/widgets/drive.coffee

```diff
@@ -51,18 +50,34 @@ getDrives = ->
 # visuals.drive('Please select a drive').then (drive) ->
 # 	console.log(drive)
 ###
-module.exports = (message = 'Select a drive') ->
-	getDrives().then (drives) ->
 
-		if _.isEmpty(drives)
-			throw new Error('No available drives')
+module.exports = (message = 'Select a drive') ->
 
-		form.ask
-			type: 'list'
-			name: 'drive'
+	getDrives().then (drives) ->
+		options =
 			message: message
+			name: 'list'
 			choices: _.map drives, (drive) ->
 				return {
 					name: "#{drive.device} (#{drive.size}) - #{drive.description}"
 					value: drive.device
 				}
+
+		ui = new UI
+			input: process.stdin
+			zoutput: process.stdout
+		list = new InquirerList(options, ui.rl)
+		scanner = new DriveScanner(getDrives, interval: 5000, drives: drives)
+
+		scanner.on 'add', (drive) ->
+			list.opt.choices.push("#{drive.device} (#{drive.size}) - #{drive.description}")
+			list.render()
+
+		scanner.on 'remove', (drive) ->
+			list.opt.choices.choices = _.filter list.opt.choices.choices, (x) ->
```

Comment:
Can you pick a more descriptive name for the predicate argument? Maybe `choice` instead of `x`?